src/ass/kfx.py

```diff
@@ -0,0 +1,237 @@
+
+    # Generate line style tags
+    eventParts: list[pyass.EventPart] = [
+        pyass.EventPart(
+            tags=[
+                *LYRICS_TAGS,
+                get_en_pos_tag(line) if line.isEN else get_romaji_pos_tag(line),
+            ],
+        )
+    ]
+
+    # Set color if there's a constant actor
+    if len(line.actorSwitches) == 0:
+        eventParts.append(pyass.EventPart(tags=actorToStyle[line.startActor]))
+
+    # Leading switch duration
+    if not line.isEN:
+        eventParts.append(pyass.EventPart(tags=[pyass.KaraokeTag(switchDuration)]))
+
+    # Generate character style tags
+    for char in line.chars:
+        eventParts.append(
+            pyass.EventPart(
+                tags=[
+                    *get_char_transform_tags(char, switchDuration, transitionDuration),
+                    *get_char_actor_tag(char, actorToStyle, switchDuration),
+                    *get_char_karaoke_tag(char),
+                ],
+                text=char.char,
+            )
+        )
+
+    # Trailing switch duration
+    if not line.isEN:
+        eventParts.append(pyass.EventPart(tags=[pyass.KaraokeTag(switchDuration)]))
+
+    return pyass.Event(
+        format=get_line_format(line),
+        style=EN_STYLE.name if line.isEN else ROMAJI_STYLE.name,
+        start=line.start - switchDuration,
+        end=line.end + switchDuration,
+        parts=eventParts,
+    )
+
+
+def get_char_syl_karaoke_tag(
+    kChar: KChar, switchDuration: timedelta, resultTag: Sequence[pyass.Tag] = []
+) -> Sequence[pyass.Tag]:
+    if kChar.karaDuration == timedelta():
+        return []
+    # Karaoke timing
+    if len(resultTag) == 0:
+        return [pyass.KaraokeTag(kChar.karaDuration)]
+    else:
+        return [
+            pyass.TransformTag(
+                start=kChar.syl.start + switchDuration,
+                end=kChar.syl.end + switchDuration,
+                to=resultTag,
+            )
+        ]
+
+
+def to_shad_event(
+    line: KLine,
+    actorToStyle: Mapping[str, Sequence[pyass.Tag]],
+    switchDuration: timedelta,
+    transitionDuration: timedelta,
+) -> pyass.Event:
+    line.calculate_char_offsets(ROMAJI_STYLE)
+    firstColor = Color.parse("&HBBC9F8&")
+    switchColor = Color.parse("&H8A78D7&")
+
+    # Generate line style tags
+    eventParts: list[pyass.EventPart] = [
+        pyass.EventPart(
+            tags=[
+                pyass.BlurEdgesTag(1),
+                pyass.AlignmentTag(Alignment.TOP_LEFT)
+                if line.isSecondary
+                else pyass.AlignmentTag(Alignment.TOP_RIGHT),
+                pyass.PositionTag(30, 30)
+                if line.isSecondary and not line.isEN
+                else pyass.PositionTag(30, 100)
+                if line.isSecondary and line.isEN
+                else pyass.PositionTag(1890, 30)
+                if not line.isSecondary and not line.isEN
+                else pyass.PositionTag(1890, 100),
```

Comment:
At this point maybe just use a conventional if-else...

Reply:
Hey okay this is just an _example_ of a custom effect with the new changes